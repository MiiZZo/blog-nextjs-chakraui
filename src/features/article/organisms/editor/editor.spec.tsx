import React from "react";
import { cleanup, fireEvent } from "@testing-library/react";
import { myRender } from "@lib/testing/render";
import Editor from "./editor";

afterEach(cleanup);

describe("editor component", () => {
  const handleSubmit = jest.fn();
  const draftData = {
    title: "",
    body: "",
    description: ""
  };

  it("should set new title, body, description", () => {
    const { asFragment, getByTestId } = myRender(
      <Editor handleSubmit={handleSubmit} draftData={draftData} />
    );
    const titleInput = getByTestId("title-input");
    const bodyInput = getByTestId("body-input");
    const descriptionInput = getByTestId("description-input");

    fireEvent.change(titleInput, { target: { value: "title" } });
    fireEvent.change(bodyInput, { target: { value: "body" } });
    fireEvent.change(descriptionInput, { target: { value: "description" } });

    expect(asFragment()).toMatchSnapshot();
  });

  it("should show a preview", () => {
    const { asFragment, getByTestId } = myRender(
      <Editor handleSubmit={handleSubmit} draftData={draftData} />
    );
    const showButton = getByTestId("show");

    fireEvent.click(showButton);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should match the shapshot", () => {
    const { asFragment } = myRender(
      <Editor handleSubmit={handleSubmit} draftData={draftData} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
