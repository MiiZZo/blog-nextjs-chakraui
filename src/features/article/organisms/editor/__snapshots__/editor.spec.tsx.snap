// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editor component should match the shapshot 1`] = `
<DocumentFragment>
  <form
    class="css-17qrq1d"
  >
    <div
      class="css-8fkoyy"
      role="group"
    >
      <label
        class="css-176s5fq"
      >
        Title
      </label>
      <div
        class="css-kjafn5"
      >
        <input
          aria-invalid="false"
          class="css-dgww3q"
          data-testid="title-input"
          name="title"
          placeholder="Title"
          value=""
        />
        <div
          class="css-1f4xxbf"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="0"
            class="css-1bfqfh5"
            data-testid="circular-indicator"
            role="progressbar"
          >
            <svg
              class="css-24ggqe"
              viewBox="55.55555555555556 55.55555555555556 111.11111111111111 111.11111111111111"
            >
              <circle
                class="css-9y0bww"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-track="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="0"
                stroke-width="11.111111111111112"
              />
              <circle
                class="css-ich4kh"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-indicator="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="314.1592653589793"
                stroke-width="11.111111111111112"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-8fkoyy"
      role="group"
    >
      <label
        class="css-176s5fq"
      >
        Description
      </label>
      <div
        class="css-kjafn5"
      >
        <input
          aria-invalid="false"
          class="css-dgww3q"
          data-testid="description-input"
          name="description"
          placeholder="Description"
          value=""
        />
        <div
          class="css-1f4xxbf"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="0"
            class="css-1bfqfh5"
            data-testid="circular-indicator"
            role="progressbar"
          >
            <svg
              class="css-24ggqe"
              viewBox="55.55555555555556 55.55555555555556 111.11111111111111 111.11111111111111"
            >
              <circle
                class="css-9y0bww"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-track="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="0"
                stroke-width="11.111111111111112"
              />
              <circle
                class="css-ich4kh"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-indicator="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="314.1592653589793"
                stroke-width="11.111111111111112"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-8fkoyy"
      role="group"
    >
      <label
        class="css-176s5fq"
      >
        Body (markdown)
      </label>
      <textarea
        aria-invalid="false"
        class="css-1tlg8wm"
        data-testid="body-input"
        name="body"
        placeholder="Body (markdown)"
      />
    </div>
    <div
      class="css-8fkoyy"
      role="group"
    >
      <button
        class="css-jifgjd"
        type="submit"
      >
        Publish article
      </button>
    </div>
    <button
      class="css-c4fa1y"
      data-testid="show"
      type="button"
    >
      Show preview
    </button>
  </form>
</DocumentFragment>
`;

exports[`editor component should set new title, body, description 1`] = `
<DocumentFragment>
  <form
    class="css-17qrq1d"
  >
    <div
      class="css-8fkoyy"
      role="group"
    >
      <label
        class="css-176s5fq"
      >
        Title
      </label>
      <div
        class="css-kjafn5"
      >
        <input
          aria-invalid="false"
          class="css-dgww3q"
          data-testid="title-input"
          name="title"
          placeholder="Title"
          value="title"
        />
        <div
          class="css-1f4xxbf"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="5"
            class="css-1bfqfh5"
            data-testid="circular-indicator"
            role="progressbar"
          >
            <svg
              class="css-24ggqe"
              viewBox="55.55555555555556 55.55555555555556 111.11111111111111 111.11111111111111"
            >
              <circle
                class="css-9y0bww"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-track="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="0"
                stroke-width="11.111111111111112"
              />
              <circle
                class="css-ich4kh"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-indicator="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="298.45130209103036"
                stroke-width="11.111111111111112"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-8fkoyy"
      role="group"
    >
      <label
        class="css-176s5fq"
      >
        Description
      </label>
      <div
        class="css-kjafn5"
      >
        <input
          aria-invalid="false"
          class="css-dgww3q"
          data-testid="description-input"
          name="description"
          placeholder="Description"
          value="description"
        />
        <div
          class="css-1f4xxbf"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="11"
            class="css-1bfqfh5"
            data-testid="circular-indicator"
            role="progressbar"
          >
            <svg
              class="css-24ggqe"
              viewBox="55.55555555555556 55.55555555555556 111.11111111111111 111.11111111111111"
            >
              <circle
                class="css-9y0bww"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-track="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="0"
                stroke-width="11.111111111111112"
              />
              <circle
                class="css-ich4kh"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-indicator="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="279.6017461694916"
                stroke-width="11.111111111111112"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-8fkoyy"
      role="group"
    >
      <label
        class="css-176s5fq"
      >
        Body (markdown)
      </label>
      <textarea
        aria-invalid="false"
        class="css-1tlg8wm"
        data-testid="body-input"
        name="body"
        placeholder="Body (markdown)"
      >
        body
      </textarea>
    </div>
    <div
      class="css-8fkoyy"
      role="group"
    >
      <button
        class="css-jifgjd"
        type="submit"
      >
        Publish article
      </button>
    </div>
    <button
      class="css-c4fa1y"
      data-testid="show"
      type="button"
    >
      Show preview
    </button>
  </form>
</DocumentFragment>
`;

exports[`editor component should show a preview 1`] = `
<DocumentFragment>
  <button
    class="css-1648ir6"
    type="button"
  >
    Close preview
  </button>
  <div
    class="css-1bxarir"
  >
    <h2
      class="css-p5x3r9"
    />
    <p
      class="css-13c355n"
    />
    <div
      class="css-j7qwjs"
    >
      <div>
        <span />
      </div>
      <div
        class="css-x60757"
      >
        <button
          class="css-1j5zsfu"
          data-testid="show"
          type="button"
        >
          Write a comment
        </button>
        <h2
          class="css-ijbsz0"
        >
          Comments - 0
        </h2>
        <div
          class="css-j7qwjs"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
